app/services/create_index_job.rb

```diff
@@ -1,14 +1,13 @@
 class CreateIndexJob < BaseIndexJob
-  def self.build_object(body:, when_completed_proc:)
-    new(book_version_id: body[:book_version_id],
-        indexing_strategy_name: body[:indexing_strategy_name],
-        when_completed_proc: when_completed_proc)
-  end
-
   def call
     index.recreate
   end
 
+  def cleanup_when_done
+    book_index = find_associated_book_index
+    book_index.mark_created
```

Comment:
Liking changing `find_associated_book_index` to `book_index_state`, which makes this method be `book_index_state.mark_created`